spec/classes/puppet_agent_windows_install_spec.rb

```diff
@@ -7,8 +7,8 @@
       is_expected.to_not contain_class('::puppet_agent::windows::install')
     }
   elsif Puppet.version >= "3.8.0"
-    {'5.1' => {:expect_arch => 'x86', :appdata => 'C:/Document and Settings/All Users/Application Data/Puppetlabs'},
-     '6.1' => {:expect_arch => 'x64', :appdata => 'C:/ProgramData/Puppetlabs'}}.each do |kernelmajversion, values|
+    {'5.1' => {:expect_arch => 'x86', :appdata => 'C:\Document and Settings\All Users\Application Data'},
```

Comment:
Fun with ruby and windows... So Ruby defines `File::SEPARATOR` to be `/` on both *nix and Windows, and Windows accepts both forward and backwards slashes. So you should be ok using `File.join`.

If you really want backslashes, you can use `File::ALT_SEPARATOR`. On Windows it's defined to be `\` and on *nix it's `nil`.

My recommendation is to be consistent with whichever you choose. For example, `File.join(`Dir.pwd`, 'bar')` can lead to mixed slashes, e.g. `C:\foo/bar`.

If you need to normalize the paths, you can do `File.expand_path(File.join(Dir.pwd, 'bar')` to produce `C:/foo/bar`